L1Trigger/L1THGCal/interface/backend/HGCalHistoSeedingImpl.h

```diff
@@ -12,6 +12,49 @@
 #include "L1Trigger/L1THGCal/interface/backend/HGCalTriggerClusterIdentificationBase.h"
 
 class HGCalHistoSeedingImpl {
+private:
+  struct Bin {
+    float sumMipPt = 0.;
+    float weighted_x = 0.;
+    float weighted_y = 0.;
+  };
+  template <typename T>
+  class HistogramT {
+  public:
+    using Data = std::vector<T>;
+    using iterator = typename Data::iterator;
+    using const_iterator = typename Data::const_iterator;
+
+  public:
+    HistogramT(unsigned bins1, unsigned bins2)
+        : bins1_(bins1), bins2_(bins2), bins_(bins1 * bins2), histogram_(bins_ * kSides_) {}
+
+    T& at(int zside, unsigned x1, unsigned x2) { return histogram_[index(zside, x1, x2)]; }
+
+    const T& at(int zside, unsigned x1, unsigned x2) const { return histogram_.at(index(zside, x1, x2)); }
```

Comment:
why does this one use the `.at()` accessor?

Reply:
The goal was to have out-of-bound checks. But there are already checks in the `index()` method, so I agree for consistency I could use `[]`